### DISCUSSION TITLE
xDrip data upload to libreview

### DISCUSSION THREAD

**Discussion by columbiecf1** [NightscoutFoundation/xDrip#2742] [Q&A]
Hello,
is there any way to send glucose data from xDrip to Libreview? In our country, when using Freestyle libre, I have to send data about glucose values to my doctor. I want to use only the xDrip app because it is the best. 

Thank you

**Comment by Navid200** [comment]
This is a simple way for creating a simple report that a doctor can  use to get what he/she needs to judge your control and provide advice for improvements:
https://navid200.github.io/xDrip/docs/Report.html

You can have followers if that's what you want.  You can use either xDrip sync or Nightscout for that:
https://navid200.github.io/xDrip/docs/Follow_page.html

**Comment by columbiecf1** [comment]
Thank you for your help. Unfortunately, doctors in the Czech Republic have to show some information from Libreview to the health insurance company. There are values that the insurance company wants to know (calculations of the Libre application).

**Reply by Navid200** [reply]
@psonnera 
Patrick, 
Can this be done?

**Reply by psonnera** [reply]
@Navid200 users should scan the sensor manually with the vendor app or reader every 8 hours to get all necessary data into LibreView.

**Reply by columbiecf1** [reply]
Yes, that's right. And in addition, there must be at least 10 scans per day (required by the insurance company). It is annoying. xDrip is cool and great in every way. It's a shame that there is no way to get the data into Librelink or Libreview. Thanks for trying to help. You're great 👍💪

**Reply by PhilTU** [reply]
A 3rd party app or service sending data into Libreview would breach the terms and conditions of the Libreview service and likely result in action by Abbott or its agents.

**Reply by Npavkov** [reply]
I use Juggluco for main NFC scan and bluetooth read of libre2 sensor, which also sends data to libreview,  and sends data to xdrip+ via nightscout so I can then calibrate the readings for the sensor and sends to my private nightscout account.

**Comment by sschamp** [comment]
Juggluco supports this:

https://github.com/search?q=repo%3Aj-kaltes%2FJuggluco%20libreview&type=code

**Reply by psonnera** [reply]
Yes. xDrip+ doesn't. xDrip+ doesn't upload to Clarity, neither to CareLink nor to LibreView (and the vendor expressed his disagreement when Diabox did, and threatened users, and this feature was removed. Nobody wants xDrip+ users to be threatened by sensors vendors).
xDrip+ uploads to Nightscout and Tidepool. Both of them are supporters of #WeAreNotWaiting. You can scan your sensor every 8 hours with the reader or the vendor app and get all necessary data into LibreView for the endo team if they don't want to use Nightscout or Tidepool. Having a new value every minute or 5 minutes with a Libre 2 clearly demonstrates you're doing something illegal with the vendor reporting system. If your endo team is allowing you to do something illegal with the apps reading your sensor and not allowing you to use Nightscout or Tidepool, they should explain why.
This is IMHO.

**Comment by tomascalta** [comment]
Hi, I am happy to help you. I am also from Czech Republic and I am using official Librelink app in conjunction with Juggluco. Juggluco can read the data from the eu fl2 sensors together with official Librelink app. Then I enabled "Librelink patched" data stream in Juggluco (in these days, "Librelink patched" data stream format is used by multiple other applications including Juggluco). In xdrip, I selected "Librelink patched" as data source and I am using both, official Librelink app and xdrip with Juggluco sitting between them. The downside is that I do need 3 apps running in the background all the time, but I found no other way to achieve this.

**Reply by PhilTU** [reply]
With EU Libre 2 you can use xDrip and OOP2 only. 

Juggluco isn't necessary. Or you can use it on its own without xDrip

**Reply by tomascalta** [reply]
He said to you very clearly that he needs to be able to upload data to libreview for his doctor.

**Reply by Navid200** [reply]
@tomascalta Please remove the all caps.
If you have a disagreement, present it logically without adding emotion to it.

**Reply by tomascalta** [reply]
@Navid200 emotions are part of everyday life. Your not gonna get away from them.

**Reply by Navid200** [reply]
That's true
Wars are fought because of emotion.
We need to control our emotions when interacting with colleagues here over the internet, or else, it will be very easy to misunderstand or offend someone.  

Phil has been assisting xDrip users here as well as on facebook for a very long time as a volunteer.  No one has the right to attack him and being emotional is not an excuse.

If you help users here and someone goes out of line and attacks you, they will get the same treatment from me as well.  I will not allow anyone to attack you either.  

The last thing I want is that people respond to each other in that manner and raise the ante.   That will change the atmosphere and will drive away those who come here to help as volunteers as well as those who come here to ask for help.    
I appreciate everyone's support and understanding.

**Reply by tomascalta** [reply]
> That can be done with periodic NFC scans by Librelink . Or just use Juggluco. I try not to build overly complex solutions.
> 
> Or just use Juggluco alone.

You are right. It is actually possible to use xdrip oop2 with Librelink app **together on the same device**, on the current sensors we have in Czech Republic (EU version), even tho I think the documentation suggests you have to uninstall Librelink app.

I still like Juggluco because it gives me ability to view raw minute by minute datapoints inside xdrip. Would be awesome to extend "raw data" option on other sources then just Librelink patched app datasource. Oop2 already has the minute by minute data.

**Reply by PhilTU** [reply]
> It is actually possible to use xdrip oop2 with Librelink app **together on the same device**

this is not supported. I know some people have it working in some form, but it is not a universal option. In particular with a sensor that has lost connection having the Librelink app trying to connect with the wrong credentials at the same time as xDrip is trying with the correct one (that it changed) is a problem.

Some people can deny Bluetooth / Location services to Librelink but I don't have that option on my phone with Librelink-GB. I can force stop it but then an NFC scan might trigger it into action again. So when troubleshooting it is best if Librelink is simply not present. Once a problem is resolved the user can try Librelink and if it breaks they know why.
